gnat2goto/driver/tree_walk.adb

```diff
@@ -2094,17 +2101,28 @@ package body Tree_Walk is
             Set_Lhs (Op_Geq, Adjusted_Value_Expr);
             Set_Type (I     => Lower_Bound,
                       Value => Bound_Type);
-            Set_Value (I     => Lower_Bound,
-                      Value => Load_Bound_In_Hex (Get_Lower_Bound (Bound_Type),
-                        Pos (Get_Width (Bound_Type))));
             Set_Rhs (Op_Geq, Adjusted_Lower_Bound);
             Set_Type (Op_Geq, Make_Bool_Type);
             Set_Lhs (Op_Leq, Adjusted_Value_Expr);
             Set_Type (I     => Upper_Bound,
                       Value => Bound_Type);
-            Set_Value (I     => Upper_Bound,
-                      Value => Load_Bound_In_Hex (Get_Upper_Bound (Bound_Type),
-                        Pos (Get_Width (Bound_Type))));
+
+            if Kind (Bound_Type) = I_Bounded_Signedbv_Type then
```

Comment:
Might be worth having a helper function that does this dispatch for us?

Reply:
Done.